ckanext/unaids/react/components/BulkFileUploaderComponent/src/App.js

```diff
@@ -40,38 +40,68 @@ export default function App(props) {
     const [networkError, setNetworkError] = useState(false);
 
     const getDefaultUploadAction = file => {
-        // disabled until we figure out a nicer algorithm
-        // const fuzzySearchResults = new Fuse(
-        //     updateResourcesOptions,
-        //     {
-        //         // https://bit.ly/3BTLnWd
-        //         includeScore: true,
-        //         findAllMatches: false,
-        //         keys: ['fileName'],
-        //         threshold: 0.3
-        //     }
-        // ).search(file.name);
-        // let defaultUploadAction = extraResource;
-        // if (fuzzySearchResults.length === 1) {
-        //     defaultUploadAction = fuzzySearchResults[0].item;
-        // } else if (fuzzySearchResults.length > 1) {
-        //     const firstResult = fuzzySearchResults[0];
-        //     const secondResult = fuzzySearchResults[1];
-        //     const scroreDiff = firstResult.score - secondResult.score;
-        //     if (scroreDiff > 0.05) {
-        //         defaultUploadAction = fuzzySearchResults[0].item;
-        //     }
-        // }
-        // return defaultUploadAction;
-        const exactMatches = updateResourcesOptions.filter(
-            x => x.fileName === file.name
-        );
-        switch (exactMatches.length) {
-            case 1:
-                return exactMatches[0];
-            default:
-                return extraResource;
+
+        function strDiff(str1, str2){
+            // Get all chars that match at the start of both strings
+            var openingMatch = "";
+            for(let x = 0; x < str1.length; x++){
+                if(str1.charAt(x) == str2.charAt(x)) openingMatch += str1.charAt(x);
+                else break;
+            }
```

Comment:
This is old school js, es6 has much nicer ways of achieving things like this using `.reduce()`
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

Reply:
Yeah I've read a bit about this and thought a bit.  It's important that I break out of the loop when chars stop matching.  I think I will leave this as it is, but thanks for flagging those map/forEach/reduce functions.